cypress/e2e/finance/funds/multiple-budgets-for-same-fund.cy.js

```diff
@@ -0,0 +1,147 @@
+import { DevTeams, TestTypes, Permissions } from '../../../support/dictionary';
+import {
+  BudgetDetails,
+  FinanceHelper,
+  FiscalYears,
+  Ledgers,
+  Funds,
+} from '../../../support/fragments/finance';
+import TopMenu from '../../../support/fragments/topMenu';
+import Users from '../../../support/fragments/users/users';
+import { StringTools, CodeTools } from '../../../support/utils';
+import { BUDGET_STATUSES } from '../../../support/constants';
+
+describe('Finance', () => {
+  describe('Funds', () => {
+    const date = new Date();
+    const code = CodeTools(4);
+    const fiscalYears = {
+      current: {
+        ...FiscalYears.getDefaultFiscalYear(),
+        code: `${code}${StringTools.randomTwoDigitNumber()}01`,
+        periodStart: new Date(date.getFullYear(), 0, 1),
+        periodEnd: new Date(date.getFullYear(), 11, 31),
+      },
+      upcoming: {
+        ...FiscalYears.getDefaultFiscalYear(),
+        code: `${code}${StringTools.randomTwoDigitNumber()}02`,
+        periodStart: new Date(date.getFullYear() + 1, 0, 1),
+        periodEnd: new Date(date.getFullYear() + 1, 11, 31),
+      },
+      next: {
+        ...FiscalYears.getDefaultFiscalYear(),
+        code: `${code}${StringTools.randomTwoDigitNumber()}03`,
+        periodStart: new Date(date.getFullYear() + 2, 0, 1),
+        periodEnd: new Date(date.getFullYear() + 2, 11, 31),
+      },
+    };
+    const ledger = { ...Ledgers.getDefaultLedger(), fiscalYearOneId: fiscalYears.current.id };
+    const fund = { ...Funds.getDefaultFund(), ledgerId: ledger.id };
+    const testData = {
+      user: {},
+    };
+
+    before('Create test data', () => {
+      cy.getAdminToken().then(() => {
+        Object.values(fiscalYears).forEach((fiscalYear) => {
+          FiscalYears.createViaApi(fiscalYear);
+        });
+        Ledgers.createViaApi(ledger);
+        Funds.createViaApi(fund);
+      });
+
+      cy.createTempUser([Permissions.uiFinanceViewEditCreateFundAndBudget.gui]).then(
+        (userProperties) => {
+          testData.user = userProperties;
+
+          cy.login(testData.user.username, testData.user.password, {
+            path: TopMenu.fundPath,
+            waiter: Funds.waitLoading,
+          });
+        },
+      );
+    });
+
+    after('Delete test data', () => {
+      cy.getAdminToken().then(() => {
+        Users.deleteViaApi(testData.user.userId);
```

Comment:
Doesn't it required to delete `FiscalYears`, `Ledgers` and `Funds`?

Reply:
Fixed